src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -932,19 +957,32 @@ static int do_list_mapped_devices()
         index++;
         continue;
       }
-      should_print = true;
-      if (cfg.snapname.empty()) {
-        cfg.snapname = "-";
+      if (f) {
+        f->open_object_section(stringify(pid).c_str());
```

Comment:
I used `pid` as object section name (and `f->open_object_section("devices")` above) to make output be similar to what krbd returns [1]. Unfortunately it generates invalid xml:
```
zhuzha:~/ceph/ceph.ci/build% ./bin/rbd-nbd --pretty-format --format xml list-mapped | xmllint -
-:2: parser error : StartTag: invalid element name
 <6298>
  ^
```
I suppose a proper solution is:
```
f->open_array_section("devices");
...
f->open_object_section("device");
f->dump_int("id", pid);
```
But then it will be incompatible with krbd. It would be nice to fix krbd, but then it will break backward compatibility...

[1] https://github.com/ceph/ceph/blob/master/src/krbd.cc#L642

Reply:
Since you have the PR to combine all the block device actions, that will allow us the opportunity to fix the XML (i.e. the old "rbd showmapped" dumps the invalid XML whereas the new "rbd device list" uses the corrected format). Therefore, I'd say you should probably do it the correct way here.

Reply:
@dillaman It looks like I misunderstood what you proposed in [1]? Because I thought the idea was to get rid of (depricate) "rbd <device> ..." commands and use for any device type what we already had for krbd (`rbd map`, `rbd unmap`, `rbd showmapped`) specifying the device type via an option.

[1] http://tracker.ceph.com/issues/20762

Reply:
@dillaman Sorry, never mind. I didn't see your comment in the PR.

Reply:
@trociny That specific tracker ticket was actually for "rbdmap" init / systemd script [1] that automatically maps images upon startup.  I thought there was a ticket for merging the rbd CLI block device actions, but I can't find it right now. Definitely good since I also want to add support for TCMU/librbd.

[1] https://github.com/ceph/ceph/blob/master/src/rbdmap